unblob/handlers/compression/lzo.py

```diff
@@ -0,0 +1,98 @@
+import io
+from typing import List, Optional
+
+from structlog import get_logger
+
+from ...models import StructHandler, ValidChunk
+
+logger = get_logger()
+
+
+F_H_FILTER = 0x00000800
+
+
+class LZOHandler(StructHandler):
+    NAME = "lzo"
+
+    YARA_RULE = r"""
+        strings:
+            $lzo_magic = { 89 4C 5A 4F 00 0D 0A 1A 0A }
+        condition:
+            $lzo_magic
+    """
+
+    C_DEFINITIONS = r"""
+        struct lzo_header_no_filter
+        {
+            char magic[9];
+            uint16 version;
+            uint16 libversion;
+            uint16 reqversion;
+            uint8 method;
+            uint8 level;
+            uint32 flags;
+            //uint32 filter;              // only if flags & F_H_FILTER
+            uint32 mode;
+            uint32 mtime;
+            uint32 gmtdiff;
+            uint8 filename_len;
+            //char filename[];
+        }
+
+        struct lzo_header_filter
+        {
+            char magic[9];
+            uint16 version;
+            uint16 libversion;
+            uint16 reqversion;
+            uint8 method;
+            uint8 level;
+            uint32 flags;
+            uint32 filter;              // only if flags & F_H_FILTER
+            uint32 mode;
+            uint32 mtime;
+            uint32 gmtdiff;
+            uint8 filename_len;
+            //char filename[];
+        }
+
+        struct lzo_size_crc {
+            uint32 original_crc;        // (CRC32 if flags & F_H_CRC32 else Adler32)
+            uint32 uncompressed_size;
+            uint32 compressed_size;
+            uint32 uncompressed_crc;
+            uint32 compressed_crc;      // (only if flags & F_ADLER32_C or flags & F_CRC32_C)
+        }
+    """
+    HEADER_STRUCT = "lzo_header"
+
+    def calculate_chunk(
+        self, file: io.BufferedIOBase, start_offset: int
+    ) -> Optional[ValidChunk]:
+
+        header = self.cparser_be.lzo_header_no_filter(file)
+
+        if header.flags & F_H_FILTER:
+            file.seek(start_offset)
+            header = self.cparser_be.lzo_header_filter(file)
+
+        filename = file.read(header.filename_len)
+        logger.debug("LZO header parsed", header=header, filename=filename)
+
+        size_crc_header = self.cparser_be.lzo_size_crc(file)
+        logger.debug("CRC header parsed", header=size_crc_header)
```

Comment:
Until now, we did not call the `self.cparser_be` directly, as far as I can tell that's what `self.parse_header` is for. I can see why can't that be used in these cases for LZO... but what do you think is there a room to refactor here?

Would not be better to call it like this?
```python
size_crc_header = self.parse_header(file=file, endian=Endian.BIG, struct_name="lzo_size_crc")
```

Reply:
No, the current implementation is better. We are using a reference to an object method. What you are proposing is called "string typing", when you are calling methods by strings, which weakens the strong type requirements of the codebase.

Reply:
Is there any IDE help for these dynamic fields like  `lzo_size_crc` BTW? It seems like `__getattribute__` magic to me which are strings too just more hidden.

Reply:
I'm not sure about this. I thought about recommending something similar when we merged the ARJ handler (see https://github.com/IoT-Inspector/unblob/pull/46/files#diff-45cd7bc944a29cf37415ad82360ef8280dd3f11d4ca2218014281fa8e5310952R100) but in the end I don't really mind us using the `cparser_be` and `cparse_le` objects when it's called for.

I'll leave the debate up to you guys, let me know if we need to change this.